test/WebSites/ControllersFromServicesWebSite/Startup.cs

```diff
@@ -25,10 +25,15 @@ public void ConfigureServices(IServiceCollection services)
                 .AddMvc()
                 .ConfigureApplicationPartManager(manager => manager.ApplicationParts.Clear())
                 .AddApplicationPart(typeof(TimeScheduleController).GetTypeInfo().Assembly)
-                .ConfigureApplicationPartManager(manager => manager.ApplicationParts.Add(new TypesPart(
-                    typeof(AnotherController),
-                    typeof(ComponentFromServicesViewComponent),
-                    typeof(InServicesTagHelper))))
+                .ConfigureApplicationPartManager(manager =>
+                {
+                    manager.ApplicationParts.Add(new TypesPart(
+                      typeof(AnotherController),
+                      typeof(ComponentFromServicesViewComponent),
+                      typeof(InServicesTagHelper)));
+
+                    manager.FeatureProviders.Add(new AssemblyMetadataReferenceFeatureProvider());
```

Comment:
In their case, adding a `ApplicationPart`  \ `AssemblyPart` in Startup should suffice:

```
.AddMvc()
.ConfigureApplicationPartManager(manager => manager.ApplicationParts.Add(new AssemblyPart(typeof(MyTagHelper).GetTypeInfo().Assembly);
```

This test was specifically trying to avoid adding the website as an `ApplicationPart` which is why I had one of these.